presto-hive/src/main/java/io/prestosql/plugin/hive/s3/PrestoS3FileSystem.java

```diff
@@ -360,10 +360,8 @@ public FSDataInputStream open(Path path, int bufferSize)
     public FSDataOutputStream create(Path path, FsPermission permission, boolean overwrite, int bufferSize, short replication, long blockSize, Progressable progress)
             throws IOException
     {
-        if ((!overwrite) && exists(path)) {
-            throw new IOException("File already exists:" + path);
-        }
-
+        // HACK: If overwrite is false then we should check if the file already exists but doing so might break S3 read-after-write consistency so skipping it.
```

Comment:
nit: remove empty line above `HACK` and add empty line after `TODO` (so it doesn't look related to `if (!stagingDirectory.exists()) {`

Reply:
How about changing this to
```java
// Ignore the overwrite flag, since Presto always writes to unique file names.
// Checking for file existence can break read-after-write consistency.
```
I don't like adding a TODO here since we are unlikely to implement this unnecessary-for-Presto behavior later.

Reply:
I have updated it.